src/Main/TextUserInterface.cpp

```diff
@@ -1290,7 +1294,7 @@ namespace VeraCrypt
 				options.Password = AskPassword (StringFormatter (_("Enter password for {0}"), wstring (*options.Path)));
 			}
 
-			if (!options.TrueCryptMode && (options.Pim < 0))
+			if (!options.TrueCryptMode /*&& (options.Pim < 0) re-ask in case of wrong value entered*/)
```

Comment:
Actually this is not the correct way to fix the issue. Instead, `options.Pim `should be set to -1 little bit below when we catch the exceptions `PasswordIncorrect `and `PasswordException` like what we do for `options.Password`.